src/app/content/search/utils.ts

```diff
@@ -149,7 +150,7 @@ export const highlightResults = (
     const hitHighlights = hit.highlight.visibleContent.map((highlightText, index) => {
       const highlights = getHighlightRanges(element, highlightText).map((range) => {
         const highlight = new Highlight(range.nativeRange, {content: range.toString()});
-        highlighter.highlight(highlight);
+        attachHighlight(highlight, highlighter);
```

Comment:
in the case of search highlights, these results will not have meaningful ids, so the error message is not really helpful

Reply:
What error message should I pass then?

Reply:
we want to be able to identify what result the problem happened with, but search results don't have a unique id. so i think the best we can do is include the search query, book, and page. maybe the index of the result in the results for the page too?

Reply:
ok, let's do this in a separate card then